apps/extension/src/core/domains/chains/types.ts

```diff
@@ -33,7 +33,7 @@ export type RequestUpsertCustomChain = {
   nativeTokenLogoUrl: string | null
   accountFormat: string | null
   subscanUrl: string | null
-  rpcs: { url: string }[]
+  rpcs: { url: string; genesisHash?: SignerPayloadGenesisHash }[]
```

Comment:
not sure if we should do this, this defines the argument for backend call which doesn't need this property

Reply:
Yeah this stood out to me too, but I couldn't think of a better place to put the genesisHash instead

Reply:
ok, in that case I think we should separate data needed for the form itself from data needed in the backend and run it through a transform. Sound ok?

Reply:
sounds great